Mathlib/Data/BitVec/Lemmas.lean

```diff
@@ -1,199 +0,0 @@
-  Fin.eq_of_val_eq (by simp [toFin_val, ofFin, toNat_ofNat, Nat.mod_eq_of_lt, i.is_lt])
-#align bitvec.to_fin_of_fin BitVec.toFin_ofFin
-
-theorem ofFin_toFin {n} (v : BitVec n) : ofFin (toFin v) = v := rfl
-#align bitvec.of_fin_to_fin BitVec.ofFin_toFin
-
-/-!
-### Distributivity of `BitVec.ofFin`
--/
-section
-variable (x y : Fin (2^w))
-
-@[simp] lemma ofFin_neg : ofFin (-x) = -(ofFin x) := by
-  ext; rw [neg_eq_zero_sub]; simp; rfl -- v4.7.0-rc1 issues remove?
-
-@[simp] lemma ofFin_and : ofFin (x &&& y) = ofFin x &&& ofFin y := by
-  ext
-  simp [HAnd.hAnd, AndOp.and, Fin.land, BitVec.and, toNat_ofFin]
-  -- v4.7.0-rc1 issues remove?
-
-@[simp] lemma ofFin_or  : ofFin (x ||| y) = ofFin x ||| ofFin y := by
-  ext
-  simp [HOr.hOr, OrOp.or, Fin.lor, BitVec.or, toNat_ofFin]
-  -- v4.7.0-rc1 issues remove?
-
-@[simp] lemma ofFin_xor : ofFin (x ^^^ y) = ofFin x ^^^ ofFin y := by
-  ext
-  simp [HXor.hXor, Xor.xor, Fin.xor, BitVec.xor, toNat_ofFin]
-  -- v4.7.0-rc1 issues remove?
-
-@[simp] lemma ofFin_mul : ofFin (x * y)   = ofFin x * ofFin y   := rfl
-
--- These should be simp, but Std's simp-lemmas do not allow this yet.
-lemma ofFin_zero : ofFin (0 : Fin (2^w)) = 0 := rfl
-lemma ofFin_one  : ofFin (1 : Fin (2^w)) = 1 := by
-  simp only [OfNat.ofNat, BitVec.ofNat, and_pow_two_is_mod]
-
-lemma ofFin_nsmul (n : ℕ) (x : Fin (2^w)) : ofFin (n • x) = n • ofFin x := rfl
-lemma ofFin_zsmul (z : ℤ) (x : Fin (2^w)) : ofFin (z • x) = z • ofFin x := rfl
-@[simp] lemma ofFin_pow (n : ℕ) : ofFin (x ^ n) = ofFin x ^ n := rfl
-
-@[simp] lemma ofFin_natCast (n : ℕ) : ofFin (n : Fin (2^w)) = n := by
-  simp only [Nat.cast, NatCast.natCast, OfNat.ofNat, BitVec.ofNat, and_pow_two_is_mod]
-  rfl
-
--- See Note [no_index around OfNat.ofNat]
-@[simp] lemma ofFin_ofNat (n : ℕ) :
-    ofFin (no_index (OfNat.ofNat n : Fin (2^w))) = OfNat.ofNat n := by
-  simp only [OfNat.ofNat, Fin.ofNat', BitVec.ofNat, and_pow_two_is_mod]
-
-end
-
-/-!
-### Distributivity of `Std.BitVec.toFin`
--/
-section
-variable (x y : BitVec w)
-
-@[simp] lemma toFin_neg : toFin (-x) = -(toFin x) := by
-  ext; rw [neg_eq_zero_sub]; simp; rfl
-  -- v4.7.0-rc1 issues remove?
-
--- These should be simp, but Std's simp-lemmas do not allow this yet.
-lemma toFin_zero : toFin (0 : BitVec w) = 0 := rfl
-lemma toFin_one  : toFin (1 : BitVec w) = 1 := by
-  rw [toFin_inj]; simp only [ofNat_eq_ofNat, ofFin_ofNat]
-
-lemma toFin_nsmul (n : ℕ) (x : BitVec w) : toFin (n • x) = n • x.toFin := rfl
-lemma toFin_zsmul (z : ℤ) (x : BitVec w) : toFin (z • x) = z • x.toFin := rfl
-@[simp] lemma toFin_pow (n : ℕ) : toFin (x ^ n) = x.toFin ^ n := rfl
-
-lemma toFin_natCast (n : ℕ) : toFin (n : BitVec w) = n := by
-  rw [toFin_inj]; simp only [ofFin_natCast]
-
-end
-
-/-!
-### `IntCast`
--/
-
--- Either of these follows trivially from the other. Which one to
--- prove is not yet clear.
-proof_wanted ofFin_intCast (z : ℤ) : ofFin (z : Fin (2^w)) = z
-
-proof_wanted toFin_intCast (z : ℤ) : toFin (z : BitVec w) = z
```

Comment:
We still want these `proof_wanted`s, right?

Reply:
No.

Reply:
I think the decision is that these can just be done downstream for now.